std/array.d

```diff
@@ -2277,48 +2277,47 @@ unittest
     shrinks the array as needed.
  +/
 void replaceInPlace(T, Range)(ref T[] array, size_t from, size_t to, Range stuff)
-    if(isDynamicArray!Range &&
-       is(ElementEncodingType!Range : T) &&
-       !is(T == const T) &&
-       !is(T == immutable T))
+    if(is(typeof(replace(array, from, to, stuff))))
 {
-    import std.algorithm : remove;
-    import std.typecons : tuple;
-
-    if (overlap(array, stuff).length)
+    static if(isDynamicArray!Range &&
+              is(Unqual!(ElementEncodingType!Range) == T) &&
+              !is(T == const T) &&
+              !is(T == immutable T) &&
```

Comment:
I think an unlisted bug exists here. If `T` is `inout` type, it would accidentally go into this optimized pass.
I'd suggest to use `isMutable!T`.